wasm.md

```diff
@@ -1448,6 +1437,58 @@ Start Function
          </moduleInst>
 ```
 
+Identifiers
+-----------
+
+Though each part of a module (each function, global, etc.) primarily has an index, the text format also allows us to refer to them by identifiers.
+We commonly want to store these optional identifiers when allocating something.
+In KWasm we store identifiers in maps from `Identifier` to `Int`, the `Int` being an index.
+This rule handles storing an `OptionalId` in the correct lookup table.
+
+```k
+    syntax Instr ::= "saveId" "(" String "," OptionalId "," Int ")"
+ // ---------------------------------------------------------------
+    rule <k> saveId (_, ID:OptionalId, _) => . ... </k>
+      requires notBool isIdentifier(ID)
+    rule <k> saveId ("type", ID:Identifier, IDX) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <typeIds> IDS => IDS [ ID <- IDX ] </typeIds>
+           ...
+         </moduleInst>
+    rule <k> saveId ("func", ID:Identifier, IDX) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <funcIds> IDS => IDS [ ID <- IDX ] </funcIds>
+           ...
+         </moduleInst>
+    rule <k> saveId ("tab", ID:Identifier, IDX) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <tabIds> IDS => IDS [ ID <- IDX ] </tabIds>
+           ...
+         </moduleInst>
+    rule <k> saveId ("mem", ID:Identifier, IDX) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <memIds> IDS => IDS [ ID <- IDX ] </memIds>
+           ...
+         </moduleInst>
+    rule <k> saveId ("glob", ID:Identifier, IDX) => . ... </k>
+         <curModIdx> CUR </curModIdx>
+         <moduleInst>
+           <modIdx> CUR </modIdx>
+           <globIds> IDS => IDS [ ID <- IDX ] </globIds>
+           ...
+         </moduleInst>
+    rule <k> saveId ("module", ID:Identifier, IDX) => . ... </k>
+         <moduleIds> IDS => IDS [ ID <- IDX ] </moduleIds>
```

Comment:
blank lines between the rules. everything runs together like this.

Reply:
Why aren't these rules just inlined into where the `saveId` is called? In every case, we already are going to the trouble of pattern matching out the `CUR` module, we might as well also do the `saveId` change as well at the same time instead of postponing it to the next step.

Reply:
Done